doc/release/upcoming_changes/16592.compatibility.rst

```diff
@@ -0,0 +1,4 @@
+float->timedelta promotion will raise a TypeError
+-------------------------------------------------
+``np.promote_types("float32", "m8")`` and other float to timedelta promotion
+which used to return "m8" (timedelta) before, will now raise a TypeError.
```

Comment:
I am wondering if we mention that it aligns it with `np.promote_types("m8", "float32")` to ease peoples mind that it really should not be a large change.  E.g. could also smuggle the word "consistently" in there ;). "Float and timedelta promotion consistently raises TypeError"?

But, this is also fine, so maybe I will just merge, or update myself later.

Reply:
thanks ! i have made a change to address this.

Reply:
Thanks Anirudh, lets put this in.